fgpyo/util/tests/test_metric.py

```diff
@@ -519,3 +522,24 @@ def test_metric_columns_out_of_order(tmp_path: Path, data_and_classes: DataBuild
     names = list(NameMetric.read(path=path))
     assert len(names) == 1
     assert names[0] == name
+
+
+def test_is_dataclass_instance() -> None:
+    """Test that is_dataclass_instance works as expected."""
+
+    assert is_dataclass_instance(dataclasses_data_and_classes.Person(name="name", age=42))
+    assert not is_dataclass_instance(attr_data_and_classes.Person(name="name", age=42))
+
+
+def test_is_attrs_instance() -> None:
+    """Test that is_attrs_instance works as expected."""
+
+    assert not is_attrs_instance(dataclasses_data_and_classes.Person(name="name", age=42))
+    assert is_attrs_instance(attr_data_and_classes.Person(name="name", age=42))
```

Comment:
I would also add a `not` test to make sure it won't return `True` on class objects instead of instances.

```
assert not is_*_instance(Person)
```

Reply:
good call!